client/checkout/utils/upe.js

```diff
@@ -279,12 +305,16 @@ export const getPaymentMethodTypes = ( paymentMethodType ) => {
 };
 
 /**
- * Returns the value of the email input on the blocks checkout page.
+ * Returns the email value from store API.
  *
- * @return {string} The value of email input.
+ * @return {string} The email value.
  */
 export const getBlocksEmailValue = () => {
-	return document.getElementById( 'email' ).value;
+	// .wcpay-payment-element container is rendered only when new payment method is selected
+	return document
+		.querySelector( '.wcpay-payment-element' )
+		?.closest( 'form' )
+		?.querySelector( 'input[type="email"]' )?.value;
```

Comment:
Up to 3 months ago the `email` field didn't have `type="email"`: https://github.com/woocommerce/woocommerce/pull/48611/files

Should we use the previous fallback, where we were getting the email field by its `id`?

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/9782/commits/4515b5156b7f4194a352ca6b1b6f0a5a6c0aa0b5